lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -271,11 +374,35 @@ class QcFlagService {
                 runNumber,
                 dplDetectorId,
             });
+            await this._resetQcFlagEffectivePeriod(newInstance);
```

Comment:
Why reset? You just inserted the flag, there is no notion of reset here. You can directly call it initialize or insert or whatever.

Reply:
refactored